src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -2725,13 +2725,14 @@ private static Type GetPSMethodProjectedType(Type type, bool isOut = false)
 
             if (type.IsByRef)
             {
-                var elementType = type.GetElementType();
+                var elementType = GetPSMethodProjectedType(type.GetElementType());
```

Comment:
Can we be sure that there is no type that will lead to infinite recursion?

Reply:
Yes, the most complex case that I can think of would be something like `ref char*`.